File: csharp/1394.find-lucky-integer-in-an-array.cs
/*
 * @lc app=leetcode id=1394 lang=csharp
 *
 * [1394] Find Lucky Integer in an Array
 */

using System.Linq;

// @lc code=start
public class Solution {
    public int FindLucky(int[] arr) {
        return arr
            .GroupBy(x => x, (x, a) => (Value: x, Count: a.Count()))
            .Where(t => t.Value == t.Count)
            .Append((Value: -1, Count: 0))
            .Max(t => t.Value);
    }
}
// @lc code=end

File: java/1394.find-lucky-integer-in-an-array.java
/*
 * @lc app=leetcode id=1394 lang=java
 *
 * [1394] Find Lucky Integer in an Array
 */

// @lc code=start
class Solution {
    public int findLucky(int[] arr) {
        int n = arr.length;
        int[] count = new int[n + 1];
        for (int x : arr) {
            if (x > n) {
                continue;
            }
            ++count[x];
        }
        for (int i = n; i >= 1; --i) {
            if (i == count[i]) {
                return i;
            }
        }
        return -1;
    }
}
// @lc code=end

